transport/src/main/java/io/netty/channel/nio/AbstractNioByteChannel.java

```diff
@@ -206,7 +207,42 @@ protected void doWrite(ChannelOutboundBuffer in) throws Exception {
                     // Break the loop and so incompleteWrite(...) is called.
                     break;
                 }
+            } else if (msg instanceof SliceableFileRegion) {
+                SliceableFileRegion region = (SliceableFileRegion) msg;
```

Comment:
I wonder if we should break each of these cases into its own private method. Folks may not be using each one of these types and JIT maybe able to optimize better if some unused code is factored out into methods.

@normanmaurer @ejona86 - wdyt?

Reply:
I'm not sure whether JVM would do better(maybe it will not inline the unused branch and make a smaller doWrite method?) but I think at least it will make the method easier to read and understand. I can do the refactoring if no objections. Thanks.

Reply:
refactoring out not-commonly called code paths can reduce the byte code size of a method and make it easier to inline ... we don't have todo that as part of this PR but in theory seems like a good thing todo.